fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/MixinPlayerManager.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.event.lifecycle;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.network.ClientConnection;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+
+import net.fabricmc.fabric.api.event.server.PlayerJoinCallback;
+
+@Mixin(PlayerManager.class)
+public abstract class MixinPlayerManager {
+	@Inject(at = @At("RETURN"), method = "onPlayerConnect")
```

Comment:
```suggestion
@Inject(method = "checkCanJoin", at = @At("HEAD"), cancellable = true)
```

Reply:
I think this should be called at return at onPlayerConnect because by then everything should be setup.

Reply:
This suggestion in theory changes the scope of the join event.

imo cancellation of connection should occur MUCH earlier than this.

Reply:
If the intent is to allow cancellation, perhaps there should be 2 events. an onPlayerJoin and a validatePlayerJoin or something